src/services/dogs.js

```diff
@@ -0,0 +1,11 @@
+import { checkError, client } from './client';
+
+export async function getDogs() {
+  const resp = await client.from('dogs').select('*').order('name');
+  return checkError(resp);
+}
+
+export async function getDogById(id) {
+  const resp = await client.from('dogs').select('*').match({ id });
```

Comment:
if you append the .single() supabase function here, then you don't have to data[0] on line 12 of Dog.js. 
.match() returns an array, and .single() should return only one row from the result. 
https://supabase.com/docs/reference/javascript/single
In this example, supabase adds a limit(1) as a contingency